Dockerfile

```diff
@@ -1,52 +1,65 @@
-# base node image
-FROM node:16-bullseye-slim as base
+FROM node:18-bullseye-slim@sha256:e2fbe082615911b184e192b05c55e7e38460a2c24c88d92e8c122ea0175fbe56 as base
 
-# set for base and all layer that inherit from it
-ENV NODE_ENV production
+ARG CURL_VERSION=7.74.0-1.3+deb11u5 \
+  OPENSSL_VERSION=1.1.1n-0+deb11u4 \
```

Comment:
Is there a reason we are not using openssl 3 instead of 1.1.1?

Reply:
I was pinning the version `apt-get` available in this image:

```sh
$ apt list -a openssl
Listing... Done
openssl/stable-security 1.1.1n-0+deb11u4 arm64
openssl/stable 1.1.1n-0+deb11u3 arm64
```

Reply:
From this output it seems to me that they basically only allow the latest version to be installed (the last two builds of the same version is listed above).

I would suggest only pinning the version 1.1.1 (if 3.0.0 is not available), because the letter is the security patch version, and if a new, critical bug comes out in openssl, it's probably best to let the apt system install the latest version that's still compatible (Also, probably they'd make it so, that 1.1.1n is unavailable, and they release a new, 1.1.1o, making this install step fail)

Reply:
Sure, so I could do this:

```
OPENSSL_VERSION=1.1.\*
CURL_VERSION=7.74.\*
TINI_VERSION=0.19.\*
```

Unless you have a better suggestion?

Reply:
If this works, I think this is a good solution 👍